kyo-core/shared/src/main/scala/kyo/sums.scala

```diff
@@ -1,25 +1,35 @@
 package kyo
 
 import kyo.core.*
-
-object Sums:
-    private object sums extends Sums[Any]
-    def apply[V]: Sums[V] = sums.asInstanceOf[Sums[V]]
+import scala.annotation.implicitNotFound
+import scala.util.NotGiven
 
 class Sums[V] extends Effect[Sums[V]]:
-    opaque type Command[T] = V
-
-    def add(v: V)(using Tag[Sums[V]]): Unit < Sums[V] =
-        this.suspend[Unit](v)
-
-    def run[T: Flat, S](v: T < (Sums[V] & S))(
-        using Tag[Sums[V]]
-    ): (Chunk[V], T) < S =
-        this.handle(handler)(Chunks.init, v)
+    type Command[T] = V
 
     private val handler =
         new ResultHandler[Chunk[V], Const[V], Sums[V], [T] =>> (Chunk[V], T), Any]:
             def done[T](st: Chunk[V], v: T) = (st, v)
             def resume[T, U: Flat, S](st: Chunk[V], command: V, k: T => U < (Sums[V] & S)) =
                 Resume(st.append(command), k(().asInstanceOf[T]))
 end Sums
+
+object Sums:
+    private object sums extends Sums[Any]
+    private def sums[V]: Sums[V] = sums.asInstanceOf[Sums[V]]
+
+    def add[V](v: V)(using Tag[Sums[V]]): Unit < Sums[V] =
+        sums[V].suspend[Unit](v)
+
+    class RunDsl[V]:
+        def apply[T: Flat, S](v: T < (Sums[V] & S))(
+            using Tag[Sums[V]]
+        ): (Chunk[V], T) < S =
+            sums[V].handle(sums[V].handler)(Chunks.init, v)
+    end RunDsl
+
+    def run[V](using
+        @implicitNotFound("Please specify the type to be handled. Example: 'Sums.run[Int](...)'") ng: NotGiven[V => Nothing]
```

Comment:
This effect is a bit different since its type parameter is invariant. I had to include this restriction otherwise Scala infers `Nothing` as `V` when it's not specified.

Reply:
Can you try `V >: Nothing`?

Reply:
Nice! I've pushed this change and it made Scala able to infer the type correctly when there's only one sum pending 👏